### ISSUE TITLE
Not working with client.authentication.k8s.io/v1 api

### ISSUE THREAD

**Issue by mbelang** [doitintl/kube-no-trouble#391]
I recently update to the latest authentication api and kubent stopped working

```bash
kubent       
11:20AM INF >>> Kube No Trouble `kubent` <<<
11:20AM INF version 0.6.0 (git sha 126e47b9b45550a45404b599c546bf911ffae571)
11:20AM INF Initializing collectors and retrieving data
11:20AM ERR Failed to initialize collector: <nil> error="failed to create client: exec plugin: invalid apiVersion \"client.authentication.k8s.io/v1\""
11:20AM ERR Failed to initialize collector: <nil> error="failed to create client: exec plugin: invalid apiVersion \"client.authentication.k8s.io/v1\""
11:20AM ERR Failed to initialize collector: <nil> error="failed to create client: exec plugin: invalid apiVersion \"client.authentication.k8s.io/v1\""
11:20AM INF Loaded ruleset name=custom.rego.tmpl
11:20AM INF Loaded ruleset name=deprecated-1-16.rego
11:20AM INF Loaded ruleset name=deprecated-1-22.rego
11:20AM INF Loaded ruleset name=deprecated-1-25.rego
11:20AM INF Loaded ruleset name=deprecated-1-26.rego
11:20AM INF Loaded ruleset name=deprecated-future.rego
```

kube config
```yaml
users:
- name: arn:aws:eks:ca-central-1:0123456789098:cluster/my-cluster
  user:
    exec:
      apiVersion: client.authentication.k8s.io/v1
      args:
      - --region
      - ca-central-1
      - eks
      - get-token
      - --cluster-name
      - my-cluster
      command: aws
      env: null
      interactiveMode: IfAvailable
      provideClusterInfo: false

```

**Comment by kppullin** [comment]
If you build from source, here's a way to make it work (roughly):

- Remove all code and files related to helm v2 as it isn't compatible with the latest go client or helm v3 libs, and appears likely to never be as it's end-of-life'd.
 ```
        deleted:    pkg/collector/helm2.go
        deleted:    pkg/collector/helm2_test.go
```

- Update libraries that are compatible with `client.authentication.k8s.io/v1`
```
# get latest k8s client
go get k8s.io/client-go
# get latest helm v3 library
go get helm.sh/helm/v3
# cleanup go.mod
go mod tidy
```

- Update `pkg/collector/cluster.go` to work with the latest k8s client library
```
git diff pkg/collector/cluster.go 
diff --git a/pkg/collector/cluster.go b/pkg/collector/cluster.go
index baa8415..a891ead 100644
--- a/pkg/collector/cluster.go
+++ b/pkg/collector/cluster.go
@@ -1,6 +1,7 @@
 package collector
 
 import (
+       "context"
        "encoding/json"
 
        "github.com/rs/zerolog/log"
@@ -113,7 +114,7 @@ func (c *ClusterCollector) Get() ([]map[string]interface{}, error) {
        for _, g := range gvrs {
                ri := c.clientSet.Resource(g)
                log.Debug().Msgf("Retrieving: %s.%s.%s", g.Resource, g.Version, g.Group)
-               rs, err := ri.List(metav1.ListOptions{})
+               rs, err := ri.List(context.Background(), metav1.ListOptions{})
                if err != nil {
```

- Build and run as normal

**Comment by stepanstipl** [comment]
Thanks @kppullin for confirming that upgrading to latest K8S `client-go`/`apimachinery` fixes this issue. I was expecting so, but haven't tested.

We have another issue #390 which is also resolved by upgrading, which, as you noted, is blocked by Helm v2. We always knew about this, but tried to keep support for it as long as we can. But it looks like the time has come to end it 🔫 . So as per #390 comment, I should have PR/nightly to test soon (there are a couple of more things to clean up in the code and docs, but essentially changes that you shared 🙏 ).

**Comment by stepanstipl** [comment]
@mbelang this should be fixed in the latest nightly build [^1]. It would be actually great if you could confirm the issue is resolved in this build. Other than that, I would like to fix one more thing (#402), and we will release it as `0.7.0`, 🤞 soon.

[^1]: https://github.com/doitintl/kube-no-trouble/releases/tag/nightly-0.6.0-15-g6424a0b

**Comment by mbelang** [comment]
@stepanstipl Working nicely :tada: .

Thanks for fixing this

**Comment by stepanstipl** [comment]
There is now [0.7.0](https://github.com/doitintl/kube-no-trouble/releases/tag/0.7.0) release containing this fix. Thanks for your help with this! 🎉

**Comment by pb-dod** [comment]
I was able to fix this issue by updating my `~/.kube/config.~` to use `client.authentication.k8s.io/v1beta1` for `apiVersion`.

**Comment by miaojiaqi** [comment]
4:19PM ERR Failed to initialize collector: <nil> error="failed to create client: exec plugin: invalid apiVersion \"client.authentication.k8s.io/v1alpha1\""
